### PR TITLE
Convert registry to k8s.gcr.io

### PR DESCRIPTION
**Pull request by thockin**
This PR was auto-generated.  Please apply human expertise to review for correctness.

Followup to https://github.com/kubernetes/kubernetes/pull/54174 and https://github.com/kubernetes/kubernetes/pull/57824

xref https://github.com/kubernetes/release/issues/281

### CODE REVIEW

File: anago [hunk 1]
```diff
@@ -1420,7 +1420,7 @@ else
   READ_RELEASE_BUCKETS+=("$RELEASE_BUCKET_USER" "$RELEASE_BUCKET_MOCK")
   GCRIO_REPO="${FLAGS_gcrio_repo:-kubernetes-release-test}"
```
Comment (ixdy) [inline anago]: this probably needs to be `gcr.io/kubernetes-release-test` instead of just `kubernetes-release-test`

File: anago [hunk 2]
```diff
@@ -118,7 +118,7 @@ PROG=${0##*/}
 #+     [--buildversion=ver]      - Override Jenkins check and set a specific
 #+                                 build version
 #+     [--gcrio_repo=repo]       - Specify the GCR repo to use.
```
Comment (ixdy) [inline anago]: arguably this flag name is also no longer really valid - it should really be `docker_registry` or something like that.

Comment (thockin) [reply anago]: PTAL - it seems to make sense now.

File: anago (hunk 1 above)
Comment (thockin) [reply anago]: done

**Comment by thockin** [comment]
Not sure about the regexes, but I will follow your guidance on this one .

File: lib/releaselib.sh [hunk 3]
```diff
@@ -909,7 +909,7 @@ release::docker::release () {
   else
     # TODO: remove this when all kubernetes releases produce a release-images
     # directory
-    [[ "$registry" =~ gcr.io/ ]] && docker_push_cmd=("$GCLOUD" "docker" "--")
```
Comment (david-mcmahon) [inline lib/releaselib.sh]: Looking at this for the first time in a while, it looks like docker_push_cmd can always be "$GCLOUD" "docker" "--", so better to just remove this and set it to that value in the declaration above.

File: lib/releaselib.sh [hunk 4]
```diff
@@ -975,7 +975,7 @@ release::docker::release_from_tarfiles () {
   local -a new_tags
   local new_tag
 
-  [[ "$registry" =~ gcr.io/ ]] && docker_push_cmd=("$GCLOUD" "docker" "--")
```
Comment (david-mcmahon) [inline lib/releaselib.sh]: Same here for docker_push_cmd.

File: lib/releaselib.sh (hunk 3 above)
Comment (thockin) [reply lib/releaselib.sh]: ACK

File: lib/releaselib.sh (hunk 4 above)
Comment (thockin) [reply lib/releaselib.sh]: ACK

**Comment by thockin** [comment]
Uggh, the way anago is set up, it builds up strings to figure out the right GCR repo, but this won't work any more.

The new staging-k8s.gcr.io is where we push images. It is literally an alias to gcr.io/google_containers (the existing repo) and is hosted in the US.

The contents of staging-k8s.gcr.io are automatically synced to {asia,eu,us)-k8s.gcr.io.

The new k8s.gcr.io will be a read-only alias to whichever regional repo is closest to you.

In the future, images will be promoted from staging to regional "prod" more explicitly and auditably.

So we need to plumb this idea throughout - the image you push may have a different name than the image you pull.  @david-mcmahon please advise?

**Comment by david-mcmahon** [comment]
Let me take this, @thockin.

File: lib/releaselib.sh [hunk 5]
```diff
@@ -966,7 +965,7 @@ release::docker::release_from_tarfiles () {
   local registry=$1
   local version=$2
   local build_output=$3
-  local docker_push_cmd=(docker)
+  local docker_push_cmd=("$GCLOUD" "docker" "--")
```
Comment (dekkagaijin) [inline lib/releaselib.sh]: This should always be safe and work regardless of gcloud version:
`local docker_push_cmd=("$GCLOUD" "docker" "-s" "$registry" "--")`

**Comment by thockin** [comment]
@david-mcmahon update?

**Comment by thockin** [comment]
ping?

**Comment by david-mcmahon** [comment]
Sorry @thockin, got lost in the pile. I will send you something today on this.

**Pull request closed without merging**